cniconf/watcher.go

```diff
@@ -78,7 +83,39 @@ func (w *watcher) onChange(version string) error {
 	return nil
 }
 
-func (w *watcher) apply(network metadata.Network) error {
+func (w *watcher) inserthostSubnetRules(host metadata.Host, content []byte) ([]byte, error) {
+	subnet, _ := host.Labels[hostSubnetLabel]
```

Comment:
@niusmallnan I just spoke with @ibuildthecloud and he came up with a totally different alternative.
So for the per-host-subnet catalog template, we will have:
```
subnet: __host_label__: io.rancher.network.per_host_subnet.subnet
rangeStart: __host_label__: io.rancher.network.per_host_subnet.range_start
```

When going through the CNI config, look at the value of the keys before generating the final CNI config, search for `__host_label__`, if found use the label following that to look up the host labels and get the value to populate it.